### PR TITLE
fix: fixed import urls for windows

### LINKED ISSUE #222: Error while running glee on Windows
Operating system: Windows 11
Node: 14
#### Describe the bug
When I try to run a project generated by glee using `npm run dev` command it shows this error:
```
Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only file and data URLs are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'c:'
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:791:11)
    at Loader.resolve (internal/modules/esm/loader.js:89:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:242:28)
    at Loader.import (internal/modules/esm/loader.js:177:28)
    at importModuleDynamically (internal/modules/esm/translators.js:114:35)
    at exports.importModuleDynamicallyCallback (internal/process/esm_loader.js:30:14)
    at file:///C:/Users/.../glee/src/lib/functions.js:27:35
    at Array.map (<anonymous>)
    at register (file:///C:/Users/.../node_modules/@asyncapi/glee/src/lib/functions.js:17:49)
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  code: 'ERR_UNSUPPORTED_ESM_URL_SCHEME'
}
⚡ Server websockets is ready to accept connections.
```
After trying to hit the endpoint with `websocat` I get this error:

```
↙ hello was received from websockets
'Hello\r\n'
x functions[operationId].run is not a function
    at file:///C:/Users/.../node_modules/@asyncapi/glee/src/lib/runtimes/js.js:14:8
    at new Promise (<anonymous>)
    at runJS (file:///C:/Users/.../bot-the-savior/node_modules/@asyncapi/glee/src/lib/runtimes/js.js:12:10)
    at triggerFunction (file:///C:/Users/.../node_modules/@asyncapi/glee/src/lib/runtimes/index.js:26:17)
    at Function.<anonymous> (file:///C:/Users/.../node_modules/@asyncapi/glee/src/index.js:70:11)
    at file:///C:/Users/.../node_modules/@asyncapi/glee/src/lib/glee.js:155:17   
    at C:\Users\...\node_modules\async\dist\async.js:1987:20
    at C:\Users\...\node_modules\async\dist\async.js:1928:13
    at replenish (C:\Users\...\node_modules\async\dist\async.js:446:21)
    at C:\Users\...\node_modules\async\dist\async.js:451:13

```
#### How to Reproduce
1) run `npx create-glee-app`
2) go to the app folder.
2) run `npm run dev`

#### Expected behaviour
It shouldn't show any error and run as demonstrated.

cc: @fmvilas @jonaslagoni

### PR DESCRIPTION
**Pull request by sudoshreyansh**
**Description**

- Fixed absolute file URLs in dynamic imports

**Related issue(s)**

Fixes #222

### CODE REVIEW

File: src/lib/functions.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-import GleeMessage from './message'
-import { GleeFunction } from './index'
-import Glee from './glee'
-import { gleeMessageToFunctionEvent } from './util'
+import { getConfigs } from './configs.js'
+import { logWarningMessage } from './logger.js'
+import GleeMessage from './message.js'
+import { GleeFunction } from './index.js'
+import Glee from './glee.js'
+import { gleeMessageToFunctionEvent } from './util.js'
```
Comment (fmvilas) [inline src/lib/functions.ts:9]:
```suggestion
import { getConfigs } from './configs.js'
import { logWarningMessage } from './logger'
import GleeMessage from './message'
import { GleeFunction } from './index'
import Glee from './glee'
import { gleeMessageToFunctionEvent } from './util'
```

Comment (fmvilas) [reply src/lib/functions.ts:9]: This one was actually my fault while merging with `master`.

File: src/lib/lifecycleEvents.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import { stat } from 'fs/promises'
 import walkdir from 'walkdir'
-import { GleeFunctionEvent, GleeFunctionReturn, GleeFunctionReturnSend } from './index'
-import { logInfoMessage } from './logger'
-import GleeMessage from './message'
-import { arrayHasDuplicates } from './util'
+import { GleeFunctionEvent, GleeFunctionReturn, GleeFunctionReturnSend } from './index.js'
+import { logInfoMessage } from './logger.js'
+import GleeMessage from './message.js'
+import { arrayHasDuplicates } from './util.js'
```
Comment (fmvilas) [inline src/lib/lifecycleEvents.ts:6]:
```suggestion
import { GleeFunctionEvent, GleeFunctionReturn, GleeFunctionReturnSend } from './index'
import { logInfoMessage } from './logger'
import GleeMessage from './message'
import { arrayHasDuplicates } from './util'
```

Comment (fmvilas) [reply src/lib/lifecycleEvents.ts:6]: Same as above

**Review by fmvilas — CHANGES_REQUESTED** [review]

File: src/lib/functions.ts [hunk 3]
```diff
@@ -1,12 +1,13 @@
 import { basename, extname, relative, join } from 'path'
 import { stat } from 'fs/promises'
 import walkdir from 'walkdir'
-import { getConfigs } from './configs'
+import { getConfigs } from './configs.js'
```
Comment (fmvilas) [inline src/lib/functions.ts:4]:
```suggestion
import { getConfigs } from './configs'
```

**Review by fmvilas — CHANGES_REQUESTED** [review]

**Comment by fmvilas** [comment]
@sudoshreyansh do you think we can easily add a test to watch for it?

**Comment by fmvilas** [comment]
I just realized we should add a test for this. This PR is failing on MacOS X and probably Linux too.

I get this error while running the `dummy` example on MacOS X:

Captura de pantalla 2022-01-17 a las 16 48 35

**Comment by sudoshreyansh** [comment]
@fmvilas the change to commonjs modules in `tsconfig.json` in the master branch fixed the issue. Now it runs fine for me in Windows Node v17.3.

But the change introduced another issue. When creating a new app using `create-glee-app`, the package.json is set to ES6 modules because of which using `require()` on the functions and lifecycles, creates an error:

image

And there also is a problem with reading the package.json when running apps created from `create-glee-app`:

image

Do I fix the latter one in this PR? And I can open a PR for the former in create-glee-app repository ?

**Comment by fmvilas** [comment]
Yeah, we have to fix this on the create-glee-app repository instead.

**Pull request closed without merging**